drivers/ethernet/eth_sam_gmac.c

```diff
@@ -2097,66 +2097,65 @@ static const struct ethernet_api eth_api = {
 #endif
 };
 
-static struct device DEVICE_NAME_GET(eth0_sam_gmac);
+static struct device DEVICE_NAME_GET(eth_sam_gmac);
 
-static void eth0_irq_config(void)
+static void eth_irq_config(void)
```

Comment:
I believe it would be better to keep the old name `eth0_irq_config`. It's unlikely that we are ever going to have SAM series SoC with two ETH interfaces, still the function is instance specific and index 0 is matching `DT_INST_0_` defines from the DTS.

Reply:
Ok, keeping the `eth0_` prefix sounds reasonable, given that these are indeed instance-specific.